packages/kit/src/runtime/client/utils.js

```diff
@@ -24,17 +24,60 @@ export function scroll_state() {
 
 /** @param {Event} event */
 export function find_anchor(event) {
-	const node = event
-		.composedPath()
-		.find((e) => e instanceof Node && e.nodeName.toUpperCase() === 'A'); // SVG <a> elements have a lowercase name
-	return /** @type {HTMLAnchorElement | SVGAElement | undefined} */ (node);
+	/** @type {HTMLAnchorElement | SVGAElement | undefined} */
+	let a;
+
+	const options = {
+		/** @type {string | null} */
+		noscroll: null,
+
+		/** @type {string | null} */
+		prefetch: null,
+
+		/** @type {string | null} */
+		reload: null
+	};
+
+	for (const element of event.composedPath()) {
+		if (!(element instanceof Element)) continue;
+
+		if (!a && element.nodeName.toUpperCase() === 'A') {
+			// SVG <a> elements have a lowercase name
+			a = /** @type {HTMLAnchorElement | SVGAElement} */ (element);
+		}
+
+		if (options.noscroll === null) {
+			options.noscroll = element.getAttribute('data-sveltekit-noscroll');
+		}
+
+		if (options.prefetch === null) {
+			options.prefetch = element.getAttribute('data-sveltekit-prefetch');
+		}
+
+		if (options.reload === null) {
+			options.reload = element.getAttribute('data-sveltekit-reload');
+		}
+	}
+
+	const url = a && new URL(a instanceof SVGAElement ? a.href.baseVal : a.href, document.baseURI);
+
+	return { a, url, options };
 }
 
-/** @param {HTMLAnchorElement | SVGAElement} node */
-export function get_href(node) {
-	return node instanceof SVGAElement
-		? new URL(node.href.baseVal, document.baseURI)
-		: new URL(node.href);
+/**
+ * @param {HTMLElement} element
+ * @param {string} name
+ */
+export function get_closest_attribute(element, name) {
```

Comment:
I don't know if the native [element.closest() API](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) would work in this case?

Reply:
Whoops, this function isn't actually used — removing it.

Do you know if `element.closest()` respects shadow DOM? We ended up having to use `event.composedPath()` because manual traversal bypasses shadow DOM nodes, annoyingly.

Reply:
Haven't tried but apparently `element.closest()` [doesn't cross shadow DOM boundaries](https://gist.github.com/pinkhominid/ac09780a5dfd356f6a5831d61c09b626)

Reply:
Yeah, just tried it and reached the same conclusion. Thanks for nothing once again, web component dweebs